ludwig/datasets/mixins/process.py

```diff
@@ -26,7 +30,13 @@ class IdentityProcessMixin:
     processed_dataset_path: str
 
     def process_downloaded_dataset(self):
-        os.rename(self.raw_dataset_path, self.processed_dataset_path)
+        protocol, _ = split_protocol(self.processed_dataset_path)
+        if protocol is not None:
+            fs = fsspec.filesystem(protocol)
+            fs.copy(self.raw_dataset_path,
```

Comment:
Maybe we can use `fsspec.mv` or `fsspec.rename` here. Otherwise the behavior is different between local and remote versions.